web-modules/javax-servlets-2/src/main/java/com/baeldung/servlets/FileUploadServlet.java

```diff
@@ -0,0 +1,53 @@
+package com.baeldung.servlets;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
+@WebServlet(name = "FileUploadServlet", urlPatterns = "/upload")
+@MultipartConfig(fileSizeThreshold = 1024 * 1024, maxFileSize = 1024 * 1024 * 5, maxRequestSize = 1024 * 1024 * 5 * 5)
+public class FileUploadServlet extends HttpServlet {
+
+    private static final long serialVersionUID = 1L;
+    private static final String UPLOAD_DIRECTORY = "uploads";
+
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        String uploadPath = getServletContext().getRealPath("") + File.separator + UPLOAD_DIRECTORY;
+        File uploadDir = new File(uploadPath);
+        if (!uploadDir.exists()) {
+            uploadDir.mkdir();
+        }
+
+        Part filePart = request.getPart("file");
+        if (filePart != null) {
+            String fileName = filePart.getSubmittedFileName();
+            String filePath = uploadPath + File.separator + fileName;
```

Comment:
I'm concerned about directory traversal here. We should either link to OWASP for more guidance or we should use more hardened code here.